R/MsExperiment.R

```diff
@@ -383,13 +404,13 @@ linkSampleData <- function(object, with = character(),
     if (!length(withIndex)) {
         link_string <- .parse_join_string(with)
         if (link_string[1L] == "sampleData") {
-            from <- paste0(link_string[1:2], collapse = ".")
+            from <- paste0(link_string[c(1, 2)], collapse = ".")
```

Comment:
Just out of curiosity, why `c()` rather than `:`?

Reply:
This was to get rid of the warning in BiocCheck that checks for presence of `:` in the code (which is strongly discouraged).

Reply:
Ah, I see - I think `:` should be ignored for iteration in case there's no element to iterate over (and use `seq_len` instead).